src/lib/data/source-text.js

```diff
@@ -76,6 +78,23 @@ export default class SourceText {
 
     this.sourceType = docSource.sourceType ? docSource.sourceType : this.sourceType
     this.tokenization = Object.assign({}, docSource.tokenization)
+
+    if (this.text.length === 0) {
+      this.detectedLang = false
+    }

```

Comment:
If a user clears text completly - the flag is set to false and detection would be done.